pinot-plugins/pinot-file-system/pinot-hdfs/src/main/java/org/apache/pinot/plugin/filesystem/HadoopPinotFS.java

```diff
@@ -97,13 +100,24 @@ public boolean copy(URI srcUri, URI dstUri)
       throws IOException {
     Path source = new Path(srcUri);
     Path target = new Path(dstUri);
-    RemoteIterator<LocatedFileStatus> sourceFiles = _hadoopFS.listFiles(source, true);
+    RemoteIterator<FileStatus> sourceFiles = _hadoopFS.listStatusIterator(source);
     if (sourceFiles != null) {
       while (sourceFiles.hasNext()) {
-        boolean succeeded =
-            FileUtil.copy(_hadoopFS, sourceFiles.next().getPath(), _hadoopFS, target, true, _hadoopConf);
-        if (!succeeded) {
-          return false;
+        FileStatus sourceFile = sourceFiles.next();
+        Path sourceFilePath = sourceFile.getPath();
+        if (sourceFile.isFile()) {
+          try {
+            FileUtil.copy(_hadoopFS, sourceFilePath, _hadoopFS, new Path(target, sourceFilePath.getName()), false,
+                _hadoopConf);
+          } catch (FileNotFoundException e) {
+            LOGGER.warn("Not found file {}, skipping copying it...", sourceFilePath, e);
+          }
+        } else if (sourceFile.isDirectory()) {
+          try {
+            copy(sourceFilePath.toUri(), new Path(target, sourceFilePath.getName()).toUri());
```

Comment:
Here the distURI is constructed by target + sourceFilePath. Can you make sure `sourceFilePath` is relative path instead of absolute path?

And if possible, it'd be good to add some tests for this method. Thanks!

Reply:
added a test .